pkg/socket/client.go

```diff
@@ -0,0 +1,62 @@
+package socket
+
+import (
+	"bufio"
+	"crypto/tls"
+	"net"
+	"time"
+)
+
+func NewSocket(config Config) *socket {
+	return &socket{
+		Config: config,
+		conn:   nil,
+	}
+}
+
+type socket struct {
+	Config
+	conn net.Conn
+}
+
+func (s *socket) Connect() (err error) {
+	if s.TlsConf == nil {
+		s.conn, err = net.DialTimeout(string(s.Network), s.Address, s.Timeout)
+	} else {
+		var d net.Dialer
+		d.Timeout = s.Timeout
+		s.conn, err = tls.DialWithDialer(&d, string(s.Network), s.Address, s.TlsConf)
+	}
+	return err
+}
+
+func (s *socket) Disconnect() error {
+	err := s.conn.Close()
+	s.conn = nil
+	return err
+}
```

Comment:
This method panics if "s.conn" is nil. I think that we shouldn't do it.

Reply:
done